test/libmpv_test.c

```diff
@@ -15,15 +15,256 @@
+            check_api_error(mpv_set_option(ctx, options[i], formats[i], &int_));
+            check_api_error(mpv_set_property(ctx, properties[i], formats[i], &int_));
+            break;
+        case MPV_FORMAT_DOUBLE:
+            check_api_error(mpv_set_option(ctx, options[i], formats[i], &double_));
+            check_api_error(mpv_set_property(ctx, properties[i], formats[i], &double_));
+            break;
+        }
+    }
+}
+
+static void test_file_loading(char *file)
+{
+    const char *cmd[] = {"loadfile", file, NULL};
+    check_api_error(mpv_command(ctx, cmd));
+    int loaded = 0;
+    int finished = 0;
+    time_t end = time(NULL) + 10;
+    while (time(NULL) < end) {
+        if (finished)
+            break;
+        mpv_event *event = mpv_wait_event(ctx, 0);
+        switch (event->event_id) {
+        case MPV_EVENT_FILE_LOADED:
+            // make sure it loads before exiting
+            loaded = 1;
+            break;
+        case MPV_EVENT_END_FILE:
+            if (loaded)
+                finished = 1;
+            break;
+        }
+    }
+    if (!finished)
+        fail("Unable to load test file!\n");
+}
+
+static void test_lavfi_complex(char *file)
+{
+    const char *cmd[] = {"loadfile", file, NULL};
+    check_api_error(mpv_command(ctx, cmd));
+    int finished = 0;
+    int loaded = 0;
+    time_t end = time(NULL) + 10;
+    while (time(NULL) < end) {
+        if (finished)
+            break;
+        mpv_event *event = mpv_wait_event(ctx, 0);
+        switch (event->event_id) {
+        case MPV_EVENT_FILE_LOADED:
+            // Add file as external and toggle lavfi-complex on.
+            if (!loaded) {
+                check_api_error(mpv_set_property_string(ctx, "external-files", file));
+                const char *add_cmd[] = {"video-add", file, "auto", NULL};
+                check_api_error(mpv_command(ctx, add_cmd));
+                check_api_error(mpv_set_property_string(ctx, "lavfi-complex", "[vid1] [vid2] vstack [vo]"));
+            }
+            loaded = 1;
+            break;
+        case MPV_EVENT_END_FILE:
+            if (loaded)
+                finished = 1;
+            break;
+        }
+    }
+    if (!finished)
+        fail("Lavfi complex failed!\n");
+}
+
+static void test_options_and_properties(void)
+{
+    // Order matters. string -> flag -> int -> double (repeat)
+    // One for set_option the other for set_property
+    const char *options[] = {
+        "screen-name",
+        "save-position-on-quit",
+        "cursor-autohide",
+        "speed",
+        NULL
+    };
+
+    const char *properties[] = {
+        "fs-screen-name",
+        "shuffle",
+        "sub-pos",
+        "window-scale",
+        NULL
+    };
+
+    // Must match above ordering.
+    enum mpv_format formats[] = {
+        MPV_FORMAT_STRING,
+        MPV_FORMAT_FLAG,
+        MPV_FORMAT_INT64,
+        MPV_FORMAT_DOUBLE,
+    };
+
+    set_options_and_properties(options, properties, formats);
```

Comment:
i dont think those properties are supposed to be set before mpv_initialize?

Reply:
It's the opposite. You can always set any option/property before `mpv_initialize`. Most things you can change afterwards, but a handful of options (like `config-dir` I think) require being set before the initialize.